Source/santad/EventProviders/SNTDeviceManagerTest.mm

```diff
@@ -0,0 +1,95 @@
+/// Copyright 2021 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+#import <DiskArbitration/DiskArbitration.h>
+#import <OCMock/OCMock.h>
+#import <XCTest/XCTest.h>
+#import <bsm/libbsm.h>
+
+#include <sys/mount.h>
+
+#import "Source/common/SNTConfigurator.h"
+#import "Source/santad/EventProviders/EndpointSecurityTestUtil.h"
+#import "Source/santad/EventProviders/SNTDeviceManager.h"
+
+@interface SNTDeviceManagerTest : XCTestCase
+@end
+
+@implementation SNTDeviceManagerTest
+
+- (void)setUp {
+  [super setUp];
+  fclose(stdout);
+}
+
+- (ESResponse *)triggerTestMount:(SNTDeviceManager *)deviceManager
+                          mockES:(MockEndpointSecurity *)mockES {
+  if (!deviceManager.subscribed) {
+    XCTestExpectation *initExpectation =
+      [self expectationWithDescription:@"Wait for SNTDeviceManager to instantiate"];
+
+    dispatch_async(dispatch_get_global_queue(QOS_CLASS_USER_INTERACTIVE, 0), ^{
+      [deviceManager listen];
+    });
+
+    dispatch_async(dispatch_get_global_queue(QOS_CLASS_USER_INTERACTIVE, 0), ^{
+      while (!deviceManager.subscribed)
```

Comment:
I don't think `[deviceManager listen]` is asynchronous at all, so this `dispatch_async` and loop aren't really necessary. (Could maybe just assert `subscribed` is set?)

Reply:
This is to enforce a hard timeout incase some future code change causes our client to never initialize correctly. I can add a comment explaining this.